File: src/main/golang/binarysearch/Sum_of_Two_Numbers.go
package binarysearch

import (
	"sort"
)

func Sum_of_Two_Numbers(nums []int, k int) bool {
	sort.Ints(nums)

	minIdx := 0
	maxIdx := len(nums) - 1

	for minIdx < maxIdx {
		sum := nums[minIdx] + nums[maxIdx]
		if (sum == k) {
			return true
		} else if(sum < k) {
			minIdx++
		} else if(sum > k) {
			maxIdx--
		}
	}
	return false
}

File: src/main/java/binarysearch/Sum_of_Two_Numbers.java
package binarysearch;

import java.util.Arrays;

public class Sum_of_Two_Numbers {

    public boolean solve(int[] nums, int k) {
        Arrays.sort(nums);

        int minIdx = 0;
        int maxIdx = nums.length - 1;

        while (minIdx < maxIdx) {
            int sum = nums[minIdx] + nums[maxIdx];
            if (sum == k) {
                return true;
            } else if (sum < k) {
                minIdx++;
            } else if (sum > k) {
                maxIdx--;
            }
        }
        return false;
    }

}
